### PR TITLE
Add docs to storage and mempool gossip mods

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

I got inspired by https://github.com/ZcashFoundation/zebra/pull/2879 and added docs to a few modules i worked on. 

This is part of https://github.com/ZcashFoundation/zebra/issues/2859

I am sorry @dconnolly , i saw the list of tasks and assigned people later on so i already had storage done. Feel free to discard it for a better description or write suggestions in this PR.

## Review

Anyone can review.

### CODE REVIEW

**Comment by teor2345** [comment]
> I am sorry @dconnolly , i saw the list of tasks and assigned people later on so i already had storage done. Feel free to discard it for a better description or write suggestions in this PR.

I gave @dconnolly some of my extra assignments, because I need to open a bunch of bug PRs today.
(And get the transaction fee into mempool storage.)

File: zebrad/src/components/mempool/gossip.rs [hunk 1]
```diff
@@ -1,4 +1,7 @@
-//! A task that gossips [`transaction::UnminedTxId`] that enter the mempool to peers.
+//! A task that gossips any [`zebra_chain::transaction::UnminedTxId`] that enters the mempool to peers.
+//!
+//! This module is just a function [`gossip_mempool_transaction_id`] that waits for mempool
+//! insertion events received in a channel and broadcast the transactions into peers.
```
Comment (conradoplg) [inline zebrad/src/components/mempool/gossip.rs]:
```suggestion
//! insertion events received in a channel and broadcasts the transactions to peers.
```

**Review by conradoplg — DISMISSED** [review]
Looks good, I just found a minor issue

**Comment by oxarbitrage** [comment]
@conradoplg can you re approve here ? I enabled auto merge.

**Review by conradoplg — APPROVED** [review]

**Pull request merged**